avocado/plugins/list.py

```diff
@@ -143,11 +171,26 @@ def _get_resolution_matrix(suite):
                     else:
                         tags_repr.append(tag)
                 tags_repr = ",".join(tags_repr)
-                test_matrix.append((type_label, runnable.uri, tags_repr))
+                test_matrix.append((runnable.kind, runnable.uri, tags_repr))
             else:
-                test_matrix.append((type_label, runnable.uri))
+                test_matrix.append((runnable.kind, runnable.uri))
         return test_matrix
 
+    @staticmethod
+    def _save_to_json(matrix, filename, verbose=False):
+        result = []
+        type_label_mapping = loader.loader.get_type_label_mapping()
+        for line in matrix:
+            if verbose:
+                result.append({'Type': type_label_mapping[line[0]],
+                               'Test': line[1],
+                               'Tags': line[2]})
+            else:
+                result.append({'Type': type_label_mapping[line[0]],
+                               'Test': line[1]})
+        with open(filename, 'a') as fp:
```

Comment:
I wouldn't use append mode here. I think that the json should be related to the updated list results and shouldn't be mixed with other results.

Reply:
Makes sense.

Reply:
1+